File: code/backend/php/tests/Unit/Core/Domain/Matches/MatchStatusesTest.php
<?php

declare(strict_types=1);

namespace Games\Paths\Tests\Unit\Core\Domain\Matches;

use Games\Paths\Core\Domain\Matches\MatchStatuses;
use PHPUnit\Framework\TestCase;

class MatchStatusesTest extends TestCase
{
    public function testAllListsTheFiveLifecycleStatuses(): void
    {
        $this->assertSame(
            ['CREATED', 'RUNNING', 'PAUSED', 'ENDED', 'GAMEOVER'],
            MatchStatuses::ALL
        );
    }

    public function testIsValid(): void
    {
        $this->assertTrue(MatchStatuses::isValid('ENDED'));
        $this->assertTrue(MatchStatuses::isValid('CREATED'));
        $this->assertFalse(MatchStatuses::isValid('BOGUS'));
        $this->assertFalse(MatchStatuses::isValid(null));
    }

    public function testIsTerminal(): void
    {
        $this->assertTrue(MatchStatuses::isTerminal('ENDED'));
        $this->assertTrue(MatchStatuses::isTerminal('GAMEOVER'));
        $this->assertFalse(MatchStatuses::isTerminal('RUNNING'));
        $this->assertFalse(MatchStatuses::isTerminal(null));
    }
}

File: code/backend/java/core/src/test/java/games/paths/core/model/match/MatchStatusesTest.java
package games.paths.core.model.match;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Unit tests for {@link MatchStatuses}.
 */
class MatchStatusesTest {

    @Test
    void allContainsTheFiveLifecycleStatuses() {
        assertEquals(List.of("CREATED", "RUNNING", "PAUSED", "ENDED", "GAMEOVER"),
                MatchStatuses.ALL);
    }

    @Test
    void isValid_acceptsKnownStatuses_rejectsOthers() {
        assertTrue(MatchStatuses.isValid("ENDED"));
        assertTrue(MatchStatuses.isValid("CREATED"));
        assertFalse(MatchStatuses.isValid("BOGUS"));
        assertFalse(MatchStatuses.isValid(null));
    }

    @Test
    void isActive_trueForCreatedRunningAndPaused() {
        assertTrue(MatchStatuses.isActive("CREATED"));
        assertTrue(MatchStatuses.isActive("RUNNING"));
        assertTrue(MatchStatuses.isActive("PAUSED"));
        assertFalse(MatchStatuses.isActive("ENDED"));
        assertFalse(MatchStatuses.isActive("GAMEOVER"));
        assertFalse(MatchStatuses.isActive(null));
    }

    @Test
    void activeAndTerminalPartitionAllStatuses() {
        assertEquals(MatchStatuses.ALL.size(),
                MatchStatuses.ACTIVE.size() + MatchStatuses.TERMINAL.size());
        MatchStatuses.ALL.forEach(s ->
                assertNotEquals(MatchStatuses.isActive(s), MatchStatuses.isTerminal(s)));
    }

    @Test
    void isTerminal_trueOnlyForEndedAndGameover() {
        assertTrue(MatchStatuses.isTerminal("ENDED"));
        assertTrue(MatchStatuses.isTerminal("GAMEOVER"));
        assertFalse(MatchStatuses.isTerminal("CREATED"));
        assertFalse(MatchStatuses.isTerminal("RUNNING"));
        assertFalse(MatchStatuses.isTerminal("PAUSED"));
        assertFalse(MatchStatuses.isTerminal(null));
    }
}
